Modules/TOF/src/TOFCheckRawsMulti.cxx

```diff
@@ -80,73 +81,79 @@ Quality TOFCheckRawsMulti::check(std::map<std::string, std::shared_ptr<MonitorOb
       //       h->SetBit(AliQAv1::GetImageBit(), kTRUE);
       //   }
       // }
-      if (h->GetEntries() == 0) {
+      if (h->GetEntries() == 0) { // Histogram is empty
         result = Quality::Medium;
         // flag = AliQAv1::kWARNING;
-      } else {
+        shifter_msg = "No counts!";
+      } else { // Histogram is non empty
         multiMean = h->GetMean();
         zeroBinIntegral = h->Integral(1, 1);
         lowMIntegral = h->Integral(1, 10);
         totIntegral = h->Integral(2, h->GetNbinsX());
 
         if (totIntegral == 0) { //if only "0 hits per event" bin is filled -> error
           if (h->GetBinContent(1) > 0) {
-            result = Quality::Medium;
+            result = Quality::Bad;
             // flag = AliQAv1::kERROR;
+            shifter_msg = "Only events at 0 filled!";
           }
         } else {
           // if (AliRecoParam::ConvertIndex(specie) == AliRecoParam::kCosmic) {
-          if (0) {
+          if (0) { // TODO: this is only for cosmics, how to check?
             if (multiMean < 10.) {
               result = Quality::Good;
+              shifter_msg = "Average within limits, OK!";
               // flag = AliQAv1::kINFO;
             } else {
               result = Quality::Medium;
+              shifter_msg = "Average outside limits!";
               // flag = AliQAv1::kWARNING;
             }
-          } else {
-            Bool_t isZeroBinContentHigh = kFALSE;
-            Bool_t isLowMultContentHigh = kFALSE;
-            Bool_t isINT7AverageLow = kFALSE;
-            Bool_t isINT7AverageHigh = kFALSE;
-
-            if (zeroBinIntegral > 0.75 * totIntegral)
-              isZeroBinContentHigh = kTRUE;
-            if (lowMIntegral > 0.75 * totIntegral)
-              isLowMultContentHigh = kTRUE;
-            if (multiMean < minTOFrawhits)
-              isINT7AverageLow = kTRUE;
-            if (multiMean > maxTOFrawhits)
-              isINT7AverageHigh = kTRUE;
+          } else { // Running with collisions
+            const Bool_t isZeroBinContentHigh = (zeroBinIntegral > fracAtZeroMult * totIntegral);
+            const Bool_t isLowMultContentHigh = (lowMIntegral > fracAtLowMult * totIntegral);
+            const Bool_t isINT7AverageLow = (multiMean < minTOFrawhits);
+            const Bool_t isINT7AverageHigh = (multiMean > maxTOFrawhits);
 
             // if (AliRecoParam::ConvertIndex(specie) == AliRecoParam::kLowMult) {
-            if (0) {
+            if (0) { // TODO: Low multiplicity running, how to check if it is pp? Probably this can be simplified in the json
```

Comment:
Good question. One way would be to add a parameter to a Check json configuration:
```
"MyCheck": {
  ...
  "checkParameters" : {
    "runType" : "cosmics"
  }
 ...
}
```
But in a longer run, I think we should consider adding this kind of metadata to each MonitorObject in framework, as soon as we are able to retrieve it. @Barthelemy what do you think?

Reply:
I was thinking of removing all the ifs from the task and simply feeding it different json files depending on the running conditions so as to entirely delegate this to the initial configuration but  suggestions are most welcome!